### DISCUSSION TITLE
Is it possible to run rustdesk server on Freetz-NG?

### DISCUSSION THREAD

**Discussion by gc-user** [Freetz-NG/freetz-ng#965] [Allgemein] [answered]
Is it possible to run rustdesk server on Freetz-NG?

I quite easily managed to set up a rustdesk server on my systems and portforwarding on the fritzbox (stock firmware). But I'd have to either have one device on all the time I want to use rustdesk or change the settings in the fritzbox to have the ports forwarded to the device I'm using.

So, I'd rather have the fritzbox run the rustdesk server. As stock firmware won't do that for me, I'm pretty sure, I was wondering if it can be done using freetz-ng?

My guess is, yes. But I wanted to make sure first and ask before diving into another new project, i.e. installing freetz-ng. Because that's the only need I'd have right now for it.

**Comment by fda77** [comment]
Like all other packages: 
- check if fritzbox is fast enough and has needed ram
- add all dependencies the programm needs to freetz, they should be listed on its website or readme
- add the package itself into make/pkgs/rustdesk/ like other

**Comment by gc-user** [comment]
Thanks for the reply. I finally got to doing some tests / dry runs, like compiling a freetz base image, and compiling rustdesk-server for mips.

I'm not sure, if adding the package to /freetz-ng/make/pkgs/rustdesk/ would work, as the package, as the name suggests, is written in rust. I assume that the freetz build process won't understand the building instructions, also, there's no .mk file, e.g.. Or am I wrong with this assumption?

Would it be possible to add the binaries to the freetz image - like adding them to /freetz-ng/build/modified/filesystem/... ?

**Comment by fda77** [comment]
I have no idea if rust is a separae compiler, but i know freetz does not has it.
You could put own files into (read .pkg) https://github.com/Freetz-NG/freetz-ng/tree/master/addon

**Comment by gc-user** [comment]
Sorry for the radio silence. Due to only having access to a MIPS Fritzbox I couldn't test rustdesk server in the first place, let alone integrating it in the Freetz image.

Now that I had access to an arm-Fritzbox and was able to test running rustdesk server on that one, I remembered this question I had, went ahead and built an Freetz image with rustdesk server files integrated.

Now I encountered the problem that the rustdesk server, when running wants to write files, sql data to be precise. As I put the rustdesk server files into /opt/ (/root/opt/rustdesk in the addon folder for building the Freetz image) and /opt seems to be read-only, I wonder what would be the best way to solve this issue.

Make /opt writable or move the rustdesk server files to another folder (before building a new Freetz image, of course)? And if the latter, which one? Apparently, the server binaries are coded in such a way to write the sql data in the same folder where the binaries are (or at least run). I thought of moving the files to /tmp/rustdesk (/root/tmp/rustdesk in the addon folder), because having sql data written in /usr/bin somehow doesn't feel right. Unless something like /usr/local/bin is possible and not read-only.

What would be the best way to proceed?

**Comment by fda77** [comment]
You could create /opt as a "link" to your usb stick. Or as an empty directory and mount it with "-o bind" to a dynamic place during build. eg with https://github.com/Freetz-NG/freetz-ng/blob/master/fwmod_custom

**Reply by gc-user** [reply, marked as answer]
I ended up putting the binaries into the /addon section of freetz-ng and created symlinks to them in the NAS section of the arm-based Fritzbox so that the system executes them in a directory where they have write access.

Thanks for the help and hints!

**Comment by fda77** [comment]
Maybe you want to share .tar archive with it? This was the origin of addon mechanic in freetz

**Reply by gc-user** [reply]
I'm not sure I understand what you're asking.
What .tar archive should I share with what?

**Reply by fda77** [reply]
A .tar to unpack into the addon directory, just in case some one else needs rust for arm

**Reply by gc-user** [reply]
Ok, understood.

I don't think that's a good idea for several reasons.

For one, for any other fritzbox than 7530, i.e. one  that actually has a useful size of NAS storage space I would not include those files in the freetz image but just upload them the the NAS storage. Otherwise one would need to build a new image for every new release of rustdesk server.
Also, I compiled the binaries myself. I don't want to be held liable for any problems / damage others might experience.
Neither do I want to "give support". Especially, since I'm no expert but a try-and-error amateur.
And lastly, although rustdesk is open source, I don't want to upload "their" files publicly.

**Reply by fda77** [reply]
Okay
